File: SharedPreferences/app/src/main/java/com/speedsol/generations/future/sharedpreferences/MainActivity.java
package com.speedsol.generations.future.sharedpreferences;

import androidx.appcompat.app.AppCompatActivity;

import android.content.SharedPreferences;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Switch;
import android.widget.TextView;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    TextView textView;
    EditText editText;
    Switch aSwitch;
    Button save, apply;

    public static final String SHARED_PREFS = "sharedprefs";
    public static final String TEXT = "text";
    public static final String SWITCH1 = "switch1";

    public String text;
    private boolean switchOnOff;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        textView = findViewById(R.id.textshow);
        editText = findViewById(R.id.textwritten);
        aSwitch = findViewById(R.id.switch1);
        apply = findViewById(R.id.apply);
        save = findViewById(R.id.save);

        apply.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                textView.setText(editText.getText().toString());
                editText.setText("");
            }
        });

        save.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                saveData();
            }
        });

        loadData();
        updateViews();
    }

    private void saveData() {
        SharedPreferences sharedPreferences = getSharedPreferences(SHARED_PREFS,MODE_PRIVATE);
        SharedPreferences.Editor editor = sharedPreferences.edit();

        editor.putString(TEXT, textView.getText().toString());
        editor.putBoolean(SWITCH1, aSwitch.isChecked());

        editor.apply();

        Toast.makeText(this, "Data Saved Successfully", Toast.LENGTH_SHORT).show();
    }

    public void loadData() {
        SharedPreferences sharedPreferences = getSharedPreferences(SHARED_PREFS, MODE_PRIVATE);
        text = sharedPreferences.getString(TEXT, "");
        switchOnOff = sharedPreferences.getBoolean(SWITCH1,false);
    }

    public void updateViews() {
        textView.setText(text);
        aSwitch.setChecked(switchOnOff);
    }
}

File: RatingBarScrima/app/src/main/java/com/speedsol/generations/future/ratingbarscrima/MainActivity.kt
package com.speedsol.generations.future.ratingbarscrima

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.RatingBar
import android.widget.TextView

class MainActivity : AppCompatActivity() {

    internal lateinit var textView: TextView
    internal lateinit var button: Button
    internal lateinit var ratingBar: RatingBar

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        textView = findViewById<View>(R.id.tvtext) as TextView
        button = findViewById<View>(R.id.btnsubmit) as Button
        ratingBar = findViewById<View>(R.id.ratingbar) as RatingBar


    }

    fun onSubmit(view: View) {
        val ratingValue = ratingBar.rating
        if (ratingValue < 2) {
            textView.text = "Rating : $ratingValue\n Very Poor"
        } else if (ratingValue >= 2 && ratingValue <= 3) {
            textView.text = "Rating : $ratingValue\n Poor"
        } else if (ratingValue >= 3 && ratingValue <= 4) {
            textView.text = "Rating : $ratingValue\n Good"
        } else if (ratingValue >= 4 && ratingValue <= 5) {
            textView.text = "Rating : $ratingValue\n Excellent"
        } else {
            textView.text = "Please Rate Our App"
        }
    }
}
